src/elli.erl

```diff
@@ -171,14 +176,14 @@ code_change(_OldVsn, State, _Extra) ->
 %%%===================================================================
 
 remove_acceptor(State, Pid) ->
-    State#state{acceptors = lists:delete(Pid, State#state.acceptors),
```

Comment:
The `code_change` callback is only called when using release upgrades. I
think an "automatic" upgrade would have to be a runtime check.

Maybe it's okay to live without.


On Fri, Jun 12, 2015 at 3:57 PM, Florian Odronitz <notifications@github.com>
wrote:

> In src/elli.erl
> <https://github.com/knutin/elli/pull/90#discussion_r32317586>:
>
> > @@ -171,14 +176,14 @@ code_change(_OldVsn, State, _Extra) ->
> >  %%%===================================================================
> >
> >  remove_acceptor(State, Pid) ->
> > -    State#state{acceptors = lists:delete(Pid, State#state.acceptors),
>
> the state could be migrated using the code_change/3 callback:
>
> % internal representation of acceptors has% changed from a list to a ETS table% so we have to migrate the state when neededcode_change(_OldVsn, State = #state{ acceptors = Acceptors }, _Extra) when is_list(Acceptors) ->
>     AcceptorsTable = ets:new(acceptors, [private, set]),
>     [ets:insert(AcceptorsTable, {Pid}) || Pid <- Acceptors],
>     {ok, State#state{ acceptors = AcceptorsTable }};
> code_change(_OldVsn, State, _Extra) ->
>     {ok, State}.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/knutin/elli/pull/90/files#r32317586>.
>